net/udp/udp_sendto_buffered.c

```diff
@@ -650,7 +652,12 @@ ssize_t psock_udp_sendto(FAR struct socket *psock, FAR const void *buf,
               goto errout_with_lock;
             }
 
-          net_lockedwait_uninterruptible(&conn->sndsem);
+          ret = net_timedwait_uninterruptible(&conn->sndsem, timeout);
+          if (ret < 0)
+            {
+              ret = -EAGAIN;
```

Comment:
What error it intends to catch? How `ECANCELED` is handled? Maybe `timeout != UINT_MAX` is missing?

Reply:
> What error it intends to catch? How `ECANCELED` is handled? Maybe `timeout != UINT_MAX` is missing?

Ensure sendto return -EAGAIN in case of timeout.
Any good suggestions for dealing with ECANCELED?
Shouldn't `net_timedwait`  be called in any timeout case？Waiting for sendbuffer release or timeout.